dot/sync/chain_sync.go

```diff
@@ -53,692 +49,709 @@ func (s chainSyncState) String() string {
+		if totalAvailable >= uint(cs.minPeers) {
 			break
 		}
-		time.Sleep(time.Millisecond * 100)
+
+		time.Sleep(time.Second)
 	}
 
-	isSyncedGauge.Set(float64(cs.state))
+	isSyncedGauge.Set(0)
+	go cs.pendingBlocks.run(cs.finalisedCh, cs.stopCh)
+	go cs.workerPool.listenForRequests(cs.stopCh)
+
+	isFarFromTarget, err := cs.isFarFromTarget()
+	if err != nil && !errors.Is(err, errNoPeerViews) {
+		panic("failing while checking target distance: " + err.Error())
+	}
 
-	pendingBlockDoneCh := make(chan struct{})
-	cs.pendingBlockDoneCh = pendingBlockDoneCh
-	go cs.pendingBlocks.run(pendingBlockDoneCh)
-	go cs.sync()
-	cs.logSyncStarted = true
-	go cs.logSyncSpeed()
+	if isFarFromTarget {
+		cs.state.Store(bootstrap)
+		go cs.bootstrapSync()
+	}
 }
 
 func (cs *chainSync) stop() {
-	if cs.pendingBlockDoneCh != nil {
-		close(cs.pendingBlockDoneCh)
+	close(cs.stopCh)
+	<-cs.workerPool.doneCh
+}
+
+func (cs *chainSync) isFarFromTarget() (bool, error) {
+	syncTarget, err := cs.getTarget()
+	if err != nil {
+		return false, fmt.Errorf("getting target: %w", err)
 	}
-	cs.cancel()
-	if cs.logSyncStarted {
-		<-cs.logSyncDone
+
+	bestBlockHeader, err := cs.blockState.BestBlockHeader()
+	if err != nil {
+		return false, fmt.Errorf("getting best block header: %w", err)
+	}
+
+	bestBlockNumber := bestBlockHeader.Number
+	isFarFromTarget := bestBlockNumber+network.MaxBlocksInResponse < syncTarget
+	return isFarFromTarget, nil
+}
+
+func (cs *chainSync) bootstrapSync() {
+	for {
+		select {
+		case <-cs.stopCh:
+			logger.Warn("ending bootstrap sync, chain sync stop channel triggered")
+			return
+		default:
+		}
+
+		// TODO: move the syncing logs to a better place
+		finalisedHeader, err := cs.blockState.GetHighestFinalisedHeader()
+		if err != nil {
+			logger.Warnf("getting highest finalized header: %w", err)
+		} else {
+			syncTarget, err := cs.getTarget()
+			if err != nil {
+				logger.Warnf("getting target: %w", err)
+			} else {
+				logger.Infof(
+					"🚣 currently syncing, %d peers connected, "+
+						"%d available workers, "+
+						"target block number %d, "+
+						"finalised block number %d with hash %s",
+					len(cs.network.Peers()),
+					cs.workerPool.totalWorkers(),
+					syncTarget, finalisedHeader.Number, finalisedHeader.Hash())
+			}
+		}
+
+		isFarFromTarget, err := cs.isFarFromTarget()
+		if err != nil && !errors.Is(err, errNoPeerViews) {
+			logger.Criticalf("ending bootstrap sync, checking target distance: %s", err)
+			return
+		}
+
+		if isFarFromTarget {
+			bestBlockHeader, err := cs.blockState.BestBlockHeader()
+			if err != nil {
+				logger.Criticalf("getting best block header: %s", err)
+				return
+			}
+
+			cs.workerPool.useConnectedPeers()
+			err = cs.requestMaxBlocksFrom(bestBlockHeader)
+			if err != nil {
+				logger.Errorf("while executing bootsrap sync: %s", err)
+			}
+		} else {
+			// we are less than 128 blocks behind the target we can use tip sync
+			cs.state.Store(tip)
```

Comment:
Can you put a comment on the `state` variable about what it is for? May be a comment for `chainSyncState`,`tip` and `bootstrap`

I am guessing that it represents whether we have synced to the tip or not. But I am not sure about it and it is not very apparent to me.